pkgs/by-name/un/unyaffs/package.nix

```
@@ -0,0 +1,33 @@
+{
+  fetchFromGitHub,
+  lib,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "unyaffs";
+  version = "0.9";
+
+  src = fetchFromGitHub {
+    owner = "whataday";
+    repo = "unyaffs";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-FcuPaxq74gVJ6UlOhFPSMtuwUAJVV/sAxAQojhmVXCs=";
+  };
+
+  preBuild = ''
+    sed -i '54i #include <sys/sysmacros.h>' unyaffs.c
+  '';
+
+  installPhase = ''
+    install -D unyaffs -t $out/bin
+  '';
+
+  meta = with lib; {
+    description = "Tool to extract files from a YAFFS2 file system image";
+    homepage = "https://github.com/whataday/unyaffs";
+    license = licenses.gpl2;
```

https://github.com/whataday/unyaffs/blob/3732be8c708f48365d8d7c65f01d4aa710d1ae47/unyaffs.c#L8
```suggestion
    license = licenses.gpl2Only;
```